src/core/net/srp_server.hpp

```diff
@@ -432,8 +434,8 @@ class Server : public InstanceLocator, private NonCopyable
          */
         const Ip6::Address *GetAddresses(uint8_t &aAddressesNum) const
         {
-            aAddressesNum = mAddresses.GetLength();
-            return mAddresses.Front();
+            aAddressesNum = static_cast<uint8_t>(OT_MIN(mAddresses.GetLength(), NumericLimits<uint8_t>::kMax));
```

Comment:
What about changing `AAddressNum` to `uint16_t`?

Reply:
Was considering it but wanted to avoid change public APIs (which may be already used by other modules/applications and require dependent changes in other repos).

We can consider changing it in a later PR (though I persoanlly prefer to minimize changes to public APIs if/when we can). We can always add new APIs but I feel would be good to minimize changing of existing APIs

Reply:
May still be a good idea to limit the number of addresses to below  or equal to 255? E.g. 8 addresses.

Reply:
Now sure if there is benefit to add a limit on max number of addresses on SRP server side? (same way that we don't limit number services a host can register and SRP protocol also does not have any limit).

The 255 here is just to keep the API the same (since it uses a `uint8_t` variable for `aAddressesNum`).

Reply:
> The 255 here is just to keep the API the same (since it uses a `uint8_t` variable for `aAddressesNum`).

The problem with the current change is that it's actually **incorrect** because `aAddressesNum` does not always get the correct number of addresses. 
Thoughts?

Reply:
We keep track of as many addresses as host has, but if there happens to be more than 255, we return the first 255 from API :)

Reply:
> We keep track of as many addresses as host has, but if there happens to be more than 255, we return the first 255 from API :)

May actually be fine, but should we document this behavior?

Reply:
I think the fact that `aAddressNum` is `uint8_t` indicate the 255 as limit.

Reply:
> I think the fact that `aAddressNum` is `uint8_t` indicate the 255 as limit.

Isn't it too implicit?

Maybe it's best that we just change the interface to `uint16_t aAddressNum`.

Thoughts? @abtink @wgtdkp

Reply:
We can consider changing the public API in a later PR (though I personally prefer to minimize changes to public APIs if/when we can). We can always add new APIs but I feel would be good to minimize changing of existing APIs.

Reply:
SGTM to leave it to another PR.